src/hooks/useHandleTransfer.tsx

```diff
@@ -308,207 +321,132 @@ async function evm(
-
-          // We increase the gas limit estimation here by a factor of 10% to account for
-          // some faulty public JSON-RPC endpoints.
-          const gasLimit = estimateGas.mul(1100).div(1000);
-          const overrides = {
-            gasLimit,
-            // We use the legacy tx envelope here to avoid triggering gas price autodetection using EIP1559 for polygon.
-            // EIP1559 is not actually implemented in polygon. The node is only API compatible but this breaks some clients
-            // like ethers when choosing fees automatically.
-            ...(chainId === CHAIN_ID_POLYGON && { type: 0 }),
-          };
+      console.log(
+        "Canonical to ",
+        THRESHOLD_GATEWAYS[recipientChain]
+          ? "Canonical"
+          : recipientChain === CHAIN_ID_ETH
+          ? "ETH"
+          : "Non-Canonical"
+      );
 
-          const tx = await L2WormholeGateway.sendTbtc(
-            amountNormalizeAmount,
-            recipientChain,
-            zeroPad(arrayify(readableTargetAddress!), 32),
-            THRESHOLD_ARBITER_FEE,
-            THRESHOLD_NONCE,
-            overrides
-          );
+      const L2WormholeGateway = new Contract(
+        sourceAddress,
+        ThresholdL2WormholeGateway,
+        signer
+      );
 
-          const receipt = await tx.wait();
+      const amountNormalizeAmount = deNormalizeAmount(
+        normalizeAmount(transferAmountParsed, decimals),
+        decimals
+      );
 
-          dispatch(
-            setTransferTx({
-              id: receipt.transactionHash,
-              block: receipt.blockNumber,
-            })
-          );
+      const estimateGas = await L2WormholeGateway.estimateGas.sendTbtc(
+        amountNormalizeAmount,
+        recipientChain,
+        recipientAddress,
+        THRESHOLD_ARBITER_FEE,
+        THRESHOLD_NONCE
+      );
 
-          enqueueSnackbar(null, {
-            content: <Alert severity="success">Transaction confirmed</Alert>,
-          });
+      // We increase the gas limit estimation here by a factor of 10% to account for
+      // some faulty public JSON-RPC endpoints.
+      const gasLimit = estimateGas.mul(1100).div(1000);
+      const overrides = {
+        gasLimit,
+        // We use the legacy tx envelope here to avoid triggering gas price autodetection using EIP1559 for polygon.
+        // EIP1559 is not actually implemented in polygon. The node is only API compatible but this breaks some clients
+        // like ethers when choosing fees automatically.
+        ...(chainId === CHAIN_ID_POLYGON && { type: 0 }),
+      };
 
-          const sequence = parseSequenceFromLogEth(
-            receipt,
-            getBridgeAddressForChain(chainId)
-          );
-          const emitterAddress = getEmitterAddressEth(
-            getTokenBridgeAddressForChain(chainId)
-          );
+      const tx = await L2WormholeGateway.sendTbtc(
+        amountNormalizeAmount,
+        recipientChain,
+        recipientAddress,
```

Comment:
Just curious, now there's not need for any type of conversion, why?

Reply:
Before, I was using the `readableTargetAddress` which was fine for EVMs but not for other chains.

`recipientAddress` it's already processed and it's always a valid 32-bytes address 💪🏼 so using this there's no need to do any conversions